### PR TITLE
Update Pin Location Assign Method, replace `free` with `pin_constraint_disabled`

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
 - [x] To address an existing issue. If so, please provide a link to the issue: EDA-2250
 - [ ] Breaking new feature. If so, please describe details in the description part.

 ### Describe the technical details
Update Pin Location Assign Method:
image
image
image

 ### Which part of the code base require a change
 
 - [x] Library: compiler, foedagcore, newproject, projnavigator
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/Main/Tasks.cpp [hunk 1]
```diff
@@ -152,7 +152,7 @@ static std::map<FOEDAG::SynthesisOptimization, const char*> synthOptMap = {
 static std::map<FOEDAG::Compiler::PinAssignOpt, const char*> pinOptMap = {
     {FOEDAG::Compiler::PinAssignOpt::Random, "random"},
     {FOEDAG::Compiler::PinAssignOpt::In_Define_Order, "in_define_order"},
-    {FOEDAG::Compiler::PinAssignOpt::Free, "free"}};
+    {FOEDAG::Compiler::PinAssignOpt::Free, "pin_constraint_disabled"}};
```
Comment (ravic-rs) [inline src/Main/Tasks.cpp:155]: Shouldn't we change PinAssignOpt too ?

**Review by ravic-rs — COMMENTED** [review]
1 comment to resolve

**Review by alain-rs — CHANGES_REQUESTED** [review]
The Tcl command need to be able to accept both options (free and pin_constraint_disabled) and do the same thing as "free".
I don't see a change of the Tcl command parser.

**Comment by KochynVolodymyr** [comment]
> The Tcl command need to be able to accept both options (free and pin_constraint_disabled) and do the same thing as "free". I don't see a change of the Tcl command parser.

@alain-rs we support both options:
image

**Review by alain-rs — APPROVED** [review]

**Pull request merged**